src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8891,83 +8838,50 @@ void WaterThermalTankData::CalcHeatPumpWaterHeater(EnergyPlusData &state, bool c
         }
         //   pass node info and simulate crankcase heater
         if (MaxSpeedNum > 0) {
-            int VSCoilNum = HeatPump.DXCoilNum;
+            int VSCoilNum = (HeatPump.bIsIHP) ? state.dataIntegratedHP->IntegratedHeatPumps(VSCoilNum).SCWHCoilNum : HeatPump.DXCoilNum;
 
-            if (HeatPump.bIsIHP) {
-                VSCoilNum = state.dataIntegratedHP->IntegratedHeatPumps(VSCoilNum).SCWHCoilIndex;
-            }
```

Comment:
Issue here for ASIHPMixedTank.idf where VSCoilNum is uninitialized (e.g., -858993460) when used at new line 8841.